go.mod

```diff
@@ -1,8 +1,8 @@
 module go.etcd.io/bbolt
 
-go 1.21
+go 1.22
```

Comment:
What is the requirement in this project to bump minimal go version to 1.22?

Reply:
Hey @crazy-max, the rationale is documented [here](https://github.com/etcd-io/etcd/issues/18443#issue-2466789888)!

Reply:
Still not sure why. It should be the minimal supported version for this module: https://go.dev/ref/mod#go-mod-file-go

> The go directive sets the minimum version of Go required to use this module.

Reply:
@ivanvc please correct me if I am wrong. 

The main idea is the support: `Each major Go release is supported until there are two newer major releases.` [1]. Thus, we are bumping the minimal required version to make sure we are not on the unsupported Go release. 

Reference:
[1] https://go.dev/doc/devel/release

Reply:
Hi @crazy-max, thanks for raising awareness of this. We have our Golang version usage documented in the etcd-io/etcd repository: https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md#golang-versions.

Reply:
Giving this a second thought. We may want to have a different policy for projects that are libraries like bbolt and raft. Should we consider staying on the oldest supported Go version until they drop support for it? @ahrtr WDYT?

Reply:
> staying on the oldest supported Go version until they drop support for it

YES, that's the policy we have been following.

Reply:
Right, my bad. I got confused with the minors.